File: src/lib/types/transversaux.ts
// Types transversaux — Phase 11

export interface SearchResult {
  entity_type: string;
  entity_id: number;
  label: string;
  sublabel: string | null;
  route: string;
}

export interface OtExportData {
  id_ordre_travail: number;
  nom_gamme: string;
  date_prevue: string;
  id_statut_ot: number;
  nom_prestataire: string | null;
  nom_localisation: string | null;
  nom_famille: string | null;
  nom_technicien: string | null;
  commentaires: string | null;
}

File: src-tauri/src/models/transversaux.rs
use serde::{Deserialize, Serialize};

/// Résultat de recherche globale
#[derive(Debug, Serialize, Deserialize)]
pub struct SearchResult {
    pub entity_type: String,
    pub entity_id: i64,
    pub label: String,
    pub sublabel: Option<String>,
    pub route: String,
}

/// Données d'export OT (fiche imprimable)
#[derive(Debug, Serialize, Deserialize)]
pub struct OtExportData {
    pub id_ordre_travail: i64,
    pub nom_gamme: String,
    pub date_prevue: String,
    pub id_statut_ot: i64,
    pub nom_prestataire: Option<String>,
    pub nom_localisation: Option<String>,
    pub nom_famille: Option<String>,
    pub nom_technicien: Option<String>,
    pub commentaires: Option<String>,
}
